### PR TITLE
fix(runtask): Fix available pool count check.

### PR DESCRIPTION
**Pull request by princerachit**
The cstor-volume-create-listcstorpoolcr-default-0.7.0 task was getting
the size of pool as actual count + 1. This is because when the list is
split there is an extra entry with content "" appended to the array.
e.g. 'a;b;' when split will give 3 values instead of two.

This was causing the pool size to appear as one size extra.
Fixed this by adding a function splitListTrim in template.go which would
first remove prefix and suffix seperator and then internally use the
same logic as that of splitList.

Signed-off-by: Prince Rachit Sinha <prince.rachit@mayadata.io>

### CODE REVIEW

File: pkg/install/v1alpha1/cstor_volume_0.7.0.go [hunk 1]
```diff
@@ -165,8 +165,13 @@ spec:
     */}}
     {{- $replicaCount := int64 .Config.ReplicaCount.value | saveAs "rc" .ListItems -}}
     {{- $poolsList := jsonpath .JsonResult "{range .items[?(@.status.phase=='Online')]}pkey=pools,{@.metadata.uid}={@.metadata.name};{end}" | trim | default "" | splitList ";" -}}
+    {{/*
+    The last item would always be an empty string so decrease the count by one
+    */}}
+    {{- $availablePools := len $poolsList -}}
```
Comment (AmitKumarDas) [inline pkg/install/v1alpha1/cstor_volume_0.7.0.go]: How about having our own splitlist that trims the delimiter from prefix and suffix before carrying out the spilt

Comment (princerachit) [reply pkg/install/v1alpha1/cstor_volume_0.7.0.go]: Sounds good. I will implement it.

Comment (princerachit) [reply pkg/install/v1alpha1/cstor_volume_0.7.0.go]: What if there is a string `';;;'` should this return `['']` or `['','']`.

File: pkg/template/template.go [hunk 2]
```diff
@@ -839,3 +840,19 @@ func AsMapOfStrings(context string, yml string, values map[string]interface{}) (
...
+// NOTE:
+//  This is intended to be used as a template function
+//
+// Example:
+//  {{- $pools =: "pool1;pool2;" | splitListTrim ";" | -}}
+//
+// Above operation would assign an array of strings containing
+// 'pool1' and 'pool2' to pools
+func splitListTrim(sep, orig string) []string {
+	processedStr := strings.TrimSuffix(strings.TrimPrefix(orig, sep), sep)
```
Comment (vishnuitta) [inline pkg/template/template.go]: does TrimRight and TrimLeft makes sense?

File: pkg/template/template_test.go [hunk 3]
```diff
@@ -2213,3 +2212,52 @@ all:
...
+		},
+		"Suffixed seperator in string": {
+			";",
+			"pool1;pool2;",
+			[]string{"pool1", "pool2"},
+		},
+		"Multiple seperators only": {
+			";",
+			";;;;",
+			[]string{"", "", ""},
```
Comment (vishnuitta) [inline pkg/template/template_test.go]: this expected result doesn't look good..
similarly, for, ";;pool1;pool2;;"

File: pkg/template/template.go (hunk 2 above)
Comment (princerachit) [reply pkg/template/template.go]: Could create it but I don't see any utility as of now. The only edge case it the right part.

Comment (vishnuitta) [reply pkg/template/template.go]: https://golang.org/pkg/strings/#TrimRight and similarly, TrimLeft

**Pull request merged**